Adding multi-stage build to the Docker recipe (#4191)
withastro/docs: src/content/docs/en/recipes/docker.mdx

COPY . .

RUN npm install

> RUN npm build

Reviewer: Pretty sure this has to stay `npm run build`. `npm build` is an internal command an not what you specified in `package.json` scripts.

```suggestion
RUN npm run build
```

}
```

> #### Multi-stage build
> Here's an example of a more advanced Dockerfile that, thanks to Docker's [multi-stage builds](https://docs.docker.com/build/building/multi-stage/), optimizes the build process for your site, sometimes reducing the build time even by minutes.

Reviewer: I think this title might be better, since we are showing an example using Node adapter
I also combined both alineas into one, which makes it a bit shorter!

```suggestion
### Multi-stage build (using SSR)

Here's an example of a more advanced Dockerfile that, thanks to Docker's [multi-stage builds](https://docs.docker.com/build/building/multi-stage/), optimizes the build process for your site by only reinstalling the npm dependencies when the source code changes. This can reduce the build time even by minutes, depending on the size of your dependencies.
```

> By utilizing multiple build stages, we ensure that the npm dependencies are not unnecessarily reinstalled when **only the source code** changes.

Reviewer: ```suggestion
```

```docker title="Dockerfile"
FROM node:lts AS base
WORKDIR /app

# By copying only the package.json and package-lock.json here, we ensure that the following `-deps` steps are independent of the source code.
# Therefore, the `-deps` steps will be skipped if only the source code changes.

> COPY package.json pacakge-lock.json ./

Reviewer: small typo
```suggestion
COPY package.json package-lock.json ./
```
